prover/src/shared_state.rs

```diff
@@ -523,6 +539,7 @@ impl SharedState {
             let task = rw.tasks.iter_mut().find(|e| e.options == task_options);
             if let Some(task) = task {
                 // found our task, update result
+                task.completed = task_result.is_ok();
```

Comment:
This makes task uncompleted when `task_result.is_err()`, but origin filter for pending tasks is `.filter(|&e| e.result.is_none())` and now is `.filter(|&e| e.completed)`.

Error results are treated as uncompleted, is this what we want?

Reply:
`task.result = Some(task_result);`, do not distinguish between Ok and Err

Reply:
Yes, before err case is also treated as done because its result is not none, so for them this completed flag does not really take effect. The node will not process err case as long as it's result is err. BTW: only a try request can clear the err result.